kedro/io/memory_data_set.py

```diff
@@ -121,3 +121,18 @@ def _exists(self) -> bool:
         if self._data is None:
             return False
         return True
+
+    def set_max_loads(self, max_loads):
+        """Set how many times this dataset can be loaded before its data being cleared to release
+        memory. Calling this method on a ``MemoryDataSet`` that already contains data has *no
+        effect*.
+
+        Args:
+            max_loads: Maximum number of times ``load`` method of the
+                data set is allowed to be invoked, before clearing the
+                data to release memory. Any number of calls is allowed
+                if the argument is not set.
+        """
+        if self._data is not None:
```

Comment:
1. Shouldn't it be `if self._data is None`?
2. If `max_loads` has been passed to the `__init__` then the counter will continue regardless.